### DISCUSSION TITLE
Backdoor writes to RTL registers

### DISCUSSION THREAD

**Discussion by raysalemi** [cocotb/cocotb#3390] [Q&A] [answered]
Hi,

We (others, actually) are working on finished the Register Access Layer (RAL) in **pyuvm**.  We need to implement a backdoor write that writes data into a register using cocotb.

Could someone point me to the documentation on how to do that?
Ray

**Comment by themperek** [comment, marked as answer]
Normally you can set internal registres same way as anything else.
Example: `dut.A.B.c  = X` would access/assign X to register c in module B that is in module A.

**Comment by raysalemi** [comment]
You mean you just write it in the Python as a normal assignment?

**Reply by themperek** [reply]
Yes. This will `Deposit` you can also `Force` https://docs.cocotb.org/en/stable/writing_testbenches.html#forcing-and-freezing-signals

**Comment by raysalemi** [comment]
Well, that was easy.

**Reply by anpingbo** [reply]
when I want to access a register in generate block(name as A), using Top.A[1].reg_m，but the name A not in the list. If there is some method to access the generate block?

**Reply by raysalemi** [reply]
I don't know the answer to that. I would open an new issue.

**Reply by themperek** [reply]
You may need to name generate block.
You can check the name by:
```python3
    for thing in Top:
        print(thing)
```
